src/app/components/event-page/event-page.component.ts

```diff
@@ -0,0 +1,132 @@
+    this.activatedRoute.params.subscribe(params => {
+      if (params && params.id) {
+        this.eventId = params.id
+      }
+    });
+   }
+
+  ngOnInit(): void {
+    this.eventService.getEvent(this.eventId).subscribe((result: Event) => {
+      this.event.id = result['result'].id;
+      this.event.categoryId = result['result'].categoryId;
+      this.event.hostId = result['result'].hostId;
+      this.event.title = result['result'].title;
+      this.event.startingDate = result['result'].startingDate;
+      this.event.endingDate = result['result'].endingDate;
+      this.event.minimumParticipants = result['result'].minimumParticipants;
+      this.event.maximumParticipants = result['result'].maximumParticipants;
+      this.event.autoCancel = result['result'].autoCancel;
+      this.event.autoJoin = result['result'].autoJoin;
+      this.event.joinDeadline = result['result'].joinDeadline;
+      this.event.fee = result['result'].fee;
+      this.event.description = result['result'].description;
+      this.event.eventLocation = result['result'].eventLocation;
+      this.event.contactEmail = result['result'].contactEmail;
+      this.event.contactPhone = result['result'].contactPhone;
+      this.event.tags = result['result'].tags;
+      this.event.backgroundImage = result['result'].backgroundImage;
+      this.intializeTagsList();
+      this.initalizeCategoryIcon();
+    });
+    
+  }
+
+  transformStringToImage(): SafeResourceUrl {
+    return this.sanitizer.bypassSecurityTrustResourceUrl(this.event.backgroundImage);
+  }
+
+  initalizeCategoryIcon(): void {
+    this.categoryService.getCategory(this.event.categoryId).subscribe((result) => {
+      this.categoryIcon = this.sanitizer.bypassSecurityTrustResourceUrl(result.result.icon);
+      console.log(result);
+    })
+   
+  }
+
+  intializeTagsList(): void {
+    this.tagsList = this.event.tags.split("*");
+  }
+
+  transformDate(date: String): string {
+    if (date) {
+      var splittedDate = date.split('T');
+    var yearMonthDayString = splittedDate[0];
+    var hourMinuteArray = splittedDate[1].split(':');
+    var hourMinuteString = hourMinuteArray[0] + ":" + hourMinuteArray[1];
+    return yearMonthDayString + " @ " + hourMinuteString; 
+    } else {
+      return '';
+    }
+  }
+
+  getMapLink(): string {
+    return "https://www.google.com/maps/place/" + this.event.eventLocation.street + "+" + this.event.eventLocation.address + "+" + this.event.eventLocation.city;
+  }
+
+  deleteEvent(): void {
+    this.openChangeRoleDialog().subscribe(result => {
+      if (result) {
+        this.eventService.deleteEvent(this.eventId).subscribe();
+        //TODO should redirect to hosted events page
+        this.router.navigate(['/administrate-categories']);
```

Comment:
Create a class or const to group all the available routes in the application 
https://medium.com/codex/how-to-share-constants-in-typescript-project-8f76a2e40352